data/src/buffer.rs

```diff
@@ -0,0 +1,389 @@
+use crate::audiosample::ChannelMap;
+use crate::buffer_ref::BufferRef;
+use crate::frame::{get_plane_size, AudioInfo, VideoInfo};
+
+/// Decoded video frame.
+///
+/// Frames are stored in native type (8/16/32-bit elements) inside
+/// a single buffer.
+/// In case of image with several components those components are stored
+/// sequentially and can be accessed in the buffer starting at
+/// corresponding component offset.
+#[derive(Clone)]
+pub struct VideoBuffer<T> {
+    info: VideoInfo,
+    pub(crate) data: BufferRef<Vec<T>>,
+    offs: Vec<usize>,
```

Comment:
Wouldn't it make sense to use `SmallVec<[usize; 4]>` or similar here instead to reduce the number of heap allocations in 99% of the cases?

Same for the other small vecs below